### ISSUE TITLE
mod_event_pusher Error 

### ISSUE THREAD

**Issue by xkawi** [esl/MongooseIM#1704]
MongooseIM version: 2.1.1
Installed from: Ubuntu Repository

Just installed latest release of MongooseIM (2.1.1) in a fresh ubuntu 16.04 ec2 instance, and was setting up the push notification module (mod_event_push), But there was an error which I couldn't tell what's the cause:

```
2018-02-08 05:47:56.865 [error] <0.2261.0>@ejabberd_hooks:run_fold1:264 {undef,[{mod_event_pusher_push_plugin_default,should_publish,[{jid,<<"USER_B">>,<<"HOST">>,<<"x86_64">>,<<"USER_B">>,<<"HOST">>,<<"x86_64">>},{jid,<<"USER">>,<<"HOST">>,<<>>,<<"USER">>,<<"HOST">>,<<>>},{xmlel,<<"message">>,[{<<"type">>,<<"chat">>},{<<"to">>,<<"USER@HOST">>}],[{xmlel,<<"body">>,[],[{xmlcdata,<<"Teat">>}]},{xmlel,<<"markable">>,[{<<"xmlns">>,<<"urn:xmpp:chat-markers:0">>}],[]},{xmlel,<<"jsondata">>,[{<<"xmlns">>,<<"urn:jsondata">>}],[{xmlcdata,<<"{\n  \"type\" : \"text\",\n  \"timestamp\" : \"08\\/02\\/2018 05:47:35\",\n  \"data\" : \"Teat\"\n}">>}]}]}],[]},{mod_event_pusher_push,do_push_event,2,[{file,"/home/vagrant/2018-01-25_16-26-49/scripts/mongooseim/deb/mongooseim/_build/prod/lib/mongooseim/src/event_pusher/mod_event_pusher_push.erl"},{line,108}]},{mod_event_pusher,'-push_event/3-lc$^0/1-0-',3,[{file,"/home/vagrant/2018-01-25_16-26-49/scripts/mongooseim/deb/mongooseim/_build/prod/lib/mongooseim/src/event_pusher/mod_event_pusher.erl"},{line,40}]},{mod_event_pusher,push_event,3,[{file,"/home/vagrant/2018-01-25_16-26-49/scripts/mongooseim/deb/mongooseim/_build/prod/lib/mongooseim/src/event_pusher/mod_event_pusher.erl"},{line,40}]},{mod_event_pusher_hook_translator,user_send_packet,4,[{file,"/home/vagrant/2018-01-25_16-26-49/scripts/mongooseim/deb/mongooseim/_build/prod/lib/mongooseim/src/event_pusher/mod_event_pusher_hook_translator.erl"},{line,65}]},{safely,apply,3,[{file,"/home/vagrant/2018-01-25_16-26-49/scripts/mongooseim/deb/mongooseim/_build/prod/lib/mongooseim/src/safely.erl"},{line,19}]},{ejabberd_hooks,hook_apply_function,5,[{file,"/home/vagrant/2018-01-25_16-26-49/scripts/mongooseim/deb/mongooseim/_build/prod/lib/mongooseim/src/ejabberd_hooks.erl"},{line,278}]},{ejabberd_hooks,run_fold1,4,[{file,"/home/vagrant/2018-01-25_16-26-49/scripts/mongooseim/deb/mongooseim/_build/prod/lib/mongooseim/src/ejabberd_hooks.erl"},{line,261}]}]}
running hook: {user_send_packet,[{jid,<<"USER_B">>,<<"HOST">>,<<"x86_64">>,<<"USER_B">>,<<"HOST">>,<<"x86_64">>},{jid,<<"USER">>,<<"HOST">>,<<>>,<<"USER">>,<<"HOST">>,<<>>},{xmlel,<<"message">>,[{<<"type">>,<<"chat">>},{<<"to">>,<<"USER@HOST">>}],[{xmlel,<<"body">>,[],[{xmlcdata,<<"Teat">>}]},{xmlel,<<"markable">>,[{<<"xmlns">>,<<"urn:xmpp:chat-markers:0">>}],[]},{xmlel,<<"jsondata">>,[{<<"xmlns">>,<<"urn:jsondata">>}],[{xmlcdata,<<"{\n  \"type\" : \"text\",\n  \"timestamp\" : \"08\\/02\\/2018 05:47:35\",\n  \"data\" : \"Hi There\"\n}">>}]}]}]}
```

Any pointer as to what could be the reason? or how to debug this error further?

Here is my ejabberd.cfg looks like:

```
{mod_push_service_mongoosepush, [{pool_name, mongoose_push_http}, {api_version, "v2"}]},

  {mod_event_pusher, [
    {backends, [
       {push, [
          {backend, mnesia},
          {wpool, [{workers, 100}]},
          {plugin_module, mod_event_pusher_push_plugin_default}
       ]}
    ]}
  ]},


  {mod_pubsub, [
    {access_createnode, pubsub_createnode},
    {ignore_pep_from_offline, false},
    {last_item_cache, true},
    {max_items_node, 1000},
    {plugins, [<<"push">>,<<"flat">>]}
  ]}
```

**Comment by Ghostw4lk** [comment]
Hey,
I configured the mod_event_pusher today and I also use version 2.1.1.

Do you configured the 'pubsub_createnode' for the mod_pubsub?
This means who is allowed to create a pubsub node.

This is my configuration for the mod_pubsub:
```xml
%%%.   ============
%%%'   ACCESS RULES

%% Only accounts of the local ejabberd server can create PubSub nodes
{access, pubsub_createnode, [{allow, local}]}.
```

My mod_pubsub configuration:
```xml
%%%.   =======
%%%'   MODULES

   {mod_pubsub, [{access_createnode, pubsub_createnode},
                 {ignore_pep_from_offline, false},
                 {last_item_cache, true},
                 {max_items_node, 1000},
                 {plugins, [<<"flat">>, <<"pep">>, <<"push">>]}
	]},
```

Last but not least, this is my mod_event_pusher configuration:
```xml
	{mod_event_pusher, [
			{backends, [
					{push, [
					{backend, mnesia},
					{wpool,  [{workers, 200}]},
					{plugin_module, mod_event_pusher_push_plugin_defaults}
				]}
			]},
			{wpool, [{workers, 200}]},
			{plugin_module, mod_push_plugin_default}
	]},
```

Hope this will help you. :)

**Comment by arcusfelis** [comment]
there is a bug in docs

```erlang
doc/modules/mod_event_pusher.md:            {plugin_module, mod_event_pusher_push_plugin_defaults}                                
doc/modules/mod_event_pusher_push.md:* **plugin_module** (atom, default: `mod_event_pusher_push_plugin_defaults`) - module implementing `mod_event_pusher_push_plugin` behaviour,
doc/modules/mod_event_pusher_push.md:            {plugin_module, mod_event_pusher_push_plugin_default}                            
src/event_pusher/mod_event_pusher_push_plugin.erl:                           mod_event_pusher_push_plugin_defaults).              
src/event_pusher/mod_event_pusher_push_plugin_defaults.erl:-module(mod_event_pusher_push_plugin_defaults). 
```

In some places mod_event_pusher_push_plugin_default is used, but should be mod_event_pusher_push_plugin_defaults.

@Ghostw4lk Be aware, that there is no module 
```erlang
{plugin_module, mod_push_plugin_default}
```





TLDR first config is missing `s`

```erlang
{mod_push_service_mongoosepush, [{pool_name, mongoose_push_http}, {api_version, "v2"}]},

  {mod_event_pusher, [
    {backends, [
       {push, [
          {backend, mnesia},
          {wpool, [{workers, 100}]},
          {plugin_module, mod_event_pusher_push_plugin_defaults} % here
       ]}
    ]}
  ]},
```

**Comment by xkawi** [comment]
@arcusfelis thanks, it works now.
